docs/cudf/source/developer_guide/cudf_pandas.md

```diff
@@ -0,0 +1,119 @@
+Final types are types for which known operations exist for converting an object of a "fast" type to a "slow" type and vice versa.
+For example, `cudf.DataFrame` can be converted to Pandas using the method `to_pandas`, and `pd.DataFrame` can be converted to cuDF using the function `cudf.from_pandas`.
+Intermediate types are the types of the results of operations invoked on final types.
+For example, `xpd.DataFrameGroupBy` is an intermediate type that will be created during a groupby operation on the final type `xpd.DataFrame`.
+
+##### Attributes and Callable Proxy Types:
+Final proxy types are typically classes or modules, both of which have attributes.
+Classes also have methods.
+These attributes and methods must be wrapped as well to support the fast-slow proxy scheme.
+
+#### Creating New Proxy Types:
+`_FinalProxy` and `_IntermediateProxy` types are created using the functions `make_final_proxy_type` and `make_intermediate_proxy` type, respectively. Creating a new final type looks like this.
+
+```python
+DataFrame = make_final_proxy_type(
+    "DataFrame",
+    cudf.DataFrame,
+    pd.DataFrame,
+    fast_to_slow=lambda fast: fast.to_pandas(),
+    slow_to_fast=cudf.from_pandas,
+)
+```
+
+### The Fallback Mechanism
+The `_fast_slow_function_call` is the all-important function that we use to call operations the fast way (using cuDF) and if that fails, the slow way (using Pandas).
+The is also known as the fallback mechanism.
+The function looks like this:
+```python
+def _fast_slow_function_call(func: Callable, *args, **kwargs):
+    try:
+        ...
+        fast_args, fast_kwargs = _fast_arg(args), _fast_arg(kwargs)
+        result = func(*fast_args, **fast_kwargs)
+        ...
+    except Exception:
+        ...
+        slow_args, slow_kwargs = _slow_arg(args), _slow_arg(kwargs)
+        result = func(*slow_args, **slow_kwargs)
+        ...
+    return _maybe_wrap_result(result, func, *args, **kwargs), fast
+```
+As you can see the function attempts to call `func` the fast way using cuDF and if any `Exception` occurs, it calls the function using Pandas.
+In essence, this `try-except` is what allows `cudf.pandas` to support 100% of the Pandas API.
+
+At the end, the function wraps the result from either path in a fast-slow proxy object, if necessary.
+
+#### Converting Proxy Objects
+Note that before the `func` is called, the proxy object and its attributes need to be converted to either their cuDF or Pandas implementations.
+This conversion is handled in the function `_transform_arg` which both `_fast_arg` and `_slow_arg` call.
+
+`_transform_arg` is a recuirsive function that will call itself depending on the type or argument passed to it (eg. `_transform_arg` is called for each element in a list of arguments).
+
+### Using Metaclasses
+`cudf.pandas` uses a [metaclass](https://docs.python.org/3/glossary.html#term-metaclass) called (`_FastSlowProxyMeta`) to dynamically find class attributes and classmethods of fast-slow proxy types.
```

Comment:
Dynamically is a bit of a loaded term here, let's avoid it. It's not obvious exactly what dynamic means in this context (for example, after https://github.com/rapidsai/cudf/pull/14534/ we no longer wait until an attribute is queried to do the lookup, it happens on type creation).
```suggestion
`cudf.pandas` uses a [metaclass](https://docs.python.org/3/glossary.html#term-metaclass) called (`_FastSlowProxyMeta`) to find class attributes and classmethods of fast-slow proxy types.
```